crates/scroll/chainspec/src/lib.rs

```diff
@@ -82,6 +86,12 @@ impl ScrollChainSpecBuilder {
         self
     }
 
+    /// Set the Scroll config.
+    pub const fn scroll_config(mut self, config: ScrollChainConfig) -> Self {
+        self.config = config;
+        self
+    }
```

Comment:
what is the thinking behind removing the `scroll_config` and having it moved to the `build`? Is this to avoid constructing a spec without the correct config?

Reply:
Yes, there's potential that we could build with a default `ScrollChainConfig`. Additionally, none of the builder arguments mutate the state of the `ScrollChainConfig`, as such it is only really relevant in the `build` method context. It yields higher quality more resilient code but also keen to hear your thoughts.

Reply:
understood, I think that makes sense to prevent constructing a chain spec without the correct config. We could also have a type state pattern here where we add a marker trait on `ScrollChainSpecBuilder<S>`, and implement the `build` method for `ScrollChainSpecBuilder<WithConfig>`. 

Although maybe that's a little heavy for the application here.

Reply:
Let's keep this simple for now and introduce the argument in `build(..)`. If this type becomes more complex then we can migrate.